### PR TITLE
HTTP APIs for state channels onchain txs

### PR DESCRIPTION
**Pull request by velzevur**
[Pivotal story](https://www.pivotaltracker.com/story/show/156295688)
HTTP API for unsigned onchain state channels' transactions
Fixed txs serialization issues

### CODE REVIEW

**Comment by velzevur** [comment]
Test failed with
```
%%% aehttp_integration_SUITE ==> {{badmatch,{error,socket_closed_remotely}},
 [{aehttp_integration_SUITE,'-populate_block/1-fun-0-',1,
                            [{file,"/home/builder/epoch/_build/test/lib/aehttp/test/aehttp_integration_SUITE.erl"},
                             {line,3852}]},
  {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
  {aehttp_integration_SUITE,populate_block,1,
                            [{file,"/home/builder/epoch/_build/test/lib/aehttp/test/aehttp_integration_SUITE.erl"},
                             {line,3850}]},
  {aehttp_integration_SUITE,add_spend_txs,0,
                            [{file,"/home/builder/epoch/_build/test/lib/aehttp/test/aehttp_integration_SUITE.erl"},
                             {line,3846}]},
  {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
  {aehttp_integration_SUITE,generic_counts_test,2,
                            [{file,"/home/builder/epoch/_build/test/lib/aehttp/test/aehttp_integration_SUITE.erl"},
                             {line,2027}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1546}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1062}]}]}
```
not related to the PR. Restarting test

File: apps/aechannel/test/aesc_test_utils.erl [hunk 1]
```diff
@@ -98,13 +98,17 @@ create_tx_spec(InitiatorPubKey, ParticipantPubKey, Spec0, State) ->
...
+      channel_reserve    => maps:get(channel_reserve, Spec),
       lock_period        => maps:get(lock_period, Spec),
       fee                => maps:get(fee, Spec),
       nonce              => maps:get(nonce, Spec)}.
 
 create_tx_default_spec(InitiatorPubKey, State) ->
     #{initiator_amount   => 50,
       participant_amount => 50,
+      push_amount        => 20,
+      channel_reserve    => 70,
```
Comment (uwiger) [inline apps/aechannel/test/aesc_test_utils.erl]: This seems like a high number – higher than the initial amounts.
> the minimum amount both peers need to maintain

The docs aren't 100% clear on whether it's _each_ peer or _both together_, but I assume that it's the former.

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 2]
```diff
@@ -997,6 +999,182 @@ nameservice_transaction_revoke(MinerAddress, MinerPubkey) ->
...
+%% GET channel_withdraw_tx unsigned transaction
+%% GET channel_close_mutual_tx unsigned transaction
+%% GET channel_close_solo unsigned transaction
+%% GET channel_slash_tx unsigned transaction
+%% GET channel_settle_tx unsigned transaction
+state_channels_onchain_transactions(_Config) ->
+    {ok, 200, _} = get_balance_at_top(),
+    {ok, 200, #{<<"pub_key">> := MinerAddress}} = get_miner_pub_key(),
+    {ok, MinerPubkey} = aec_base58c:safe_decode(account_pubkey, MinerAddress),
+    ParticipantPubkey = random_hash(),
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: Strictly speaking, the `channel_create_tx` should be signed by both parties (although this is probably not checked yet).

Comment (velzevur) [reply apps/aehttp/test/aehttp_integration_SUITE.erl]: These are the unsigned txs, so signing here

File: apps/aechannel/test/aesc_test_utils.erl (hunk 1 above)
Comment (velzevur) [reply apps/aechannel/test/aesc_test_utils.erl]: I used the [code](https://github.com/aeternity/epoch/blob/PT-156295688_state-channels_onchaintx_HTTP_API/apps/aechannel/src/aesc_create_tx.erl#L273-L277) as an reference. According to it - this amount should be greater than the initiator.

Comment (velzevur) [reply apps/aechannel/test/aesc_test_utils.erl]: Ah, wrong link - link above was for `push_amount`. The `channel_reserve` must be higher by both of them - (correct link)[https://github.com/aeternity/epoch/blob/PT-156295688_state-channels_onchaintx_HTTP_API/apps/aechannel/src/aesc_create_tx.erl#L279-L288]

**Pull request merged**